service/cloudconfig/service.go

```diff
@@ -167,6 +173,14 @@ func (c CloudConfig) NewWorkerCloudConfig(customObject providerv1alpha1.AzureCon
 	return newCloudConfig(k8scloudconfig.WorkerTemplate, params)
 }
 
+func (c CloudConfig) getEncryptionkey(customObject providerv1alpha1.AzureConfig) (string, error) {
+	cluster, err := c.randomkeysSearcher.SearchCluster(key.ClusterID(customObject))
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+	return string(cluster.APIServerEncryptionKey), nil
+}
```

Comment:
I don't see a reason to extract that method, but when it feels cleaner to you that's ok.